examples/message-store-listener/message_store_listener.bal

```diff
@@ -0,0 +1,36 @@
+import ballerina/messaging;
+import ballerina/log;
+
+// Create the main message store
+messaging:Store msgStore = new messaging:InMemoryMessageStore();
+
+// Create the dead-letter message store
+messaging:Store dlStore = new messaging:InMemoryMessageStore();
+
+// Create the message store listener
+listener messaging:StoreListener msgStoreListener = new (msgStore, {
+    pollingInterval: 10,
+    maxRetries: 2,
+    retryInterval: 2,
+    deadLetterStore: dlStore
+});
+
+// Create the service to process messages
+service on msgStoreListener {
+    isolated remote function onMessage(anydata payload) returns error? {
+        log:printInfo("message received", payload = payload);
+
+        // Simulate error scenario
+        if payload == "error" {
+            return error("Simulated processing error");
+        }
+    }
+}
+
+public function main() returns error? {
```

Comment:
Why do we need this section ? IMO listener-service part is enough in the `store-listener` example.